### PR TITLE
New function: replicate!

### PR DESCRIPTION
**Pull request by Tortar**
Implemented the function described in #809. 

I decided to include it in the sample.jl file since this redefinition of `deepcopy` should benefit also `sample!`.

This still needs some tests. Will also benchmark the benefit for `sample!`.

### CODE REVIEW

**Comment by Tortar** [comment]
Running this:

using Agents, BenchmarkTools

mutable struct FakeAgent <: AbstractAgent
    id::Int
end

```julia
function fake_model(; nagents, scheduler)
    model = StandardABM(FakeAgent; scheduler)
    for i in 1:nagents
        add_agent!(model)
    end
    model
end

model = fake_model(; nagents = 1000, scheduler=Schedulers.fastest)

@benchmark sample!($model, 100000)
@benchmark sample!($model, 10000)
@benchmark sample!($model, 1000)
```

you can see that `sample!` is now 3x faster!

**Review by Datseris — APPROVED** [review]
seems good, i'd say up minor version and tag new release.

BTW, i was always thinking that some kind of model with `sample!` should have been in the ABM cvomparison framework since we have a simpler and more well thought out and almost surely faster API. Maybe you can open an issue in the ABGM comparison repo asking for a model with sample. Or maybe we just use the evolution model we have in the example zoo

**Comment by Tortar** [comment]
yes, I think it is a good idea, but at the same time I'm not sure if an implementation of a similar model is ready-made in the other frameworks

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # main
 
+- New function `replicate!` allows to create a new instance of a given agent at the same position with the possibility to specify some
```
Comment (Datseris) [inline CHANGELOG.md:3]: Should have been a `# v5.17` here, you can push it to master directly

Comment (Tortar) [reply CHANGELOG.md:3]: ops sure :-)

**Pull request merged**